packages/vite/src/node/ssr/ssrExternal.ts

```diff
@@ -36,6 +36,7 @@ export function resolveSSRExternal(
   }
 
   for (const id of deps) {
+    if (id === pkg.name) continue
```

Comment:
Is this change needed for the PR? Could it happen that the user names its pkg.name with the same name as one of its dependencies?

Reply:
Yes because `deps` includes `knownImports` which includes `Object.keys(require('node_modules/.vite/_metadata.json').optimized`. This can (and does) lead to `id === pkg.name` which results in an infinite recursion (this has actually happened).

Reply:
Is there a way to add a test case for this so others are not tempted to remove it?

Reply:
I'll have a look at it but I'm not sure there is a non-convoluted way to test this.

Reply:
@brillout would you check if this PR solves the infinite loop in a more robust way https://github.com/vitejs/vite/pull/2636 ?

Reply:
@patak-js Good point, added a comment over there. Sorry for the late reply - was busy.

Reply:
@patak-js Btw. is there anything I can do to increase the likelyhood for https://github.com/vitejs/vite/pull/2519 to get merged? I'm a bit worried it doesn't get meged; it's critical for vite-plugin-ssr.

Reply:
We already requested a review from Evan, we need his input on that one

Reply:
@patak-js I'm thinking that with https://github.com/vitejs/vite/pull/2636#issuecomment-812201225 we wouldn't really need a test anymore as it makes the handling of inifinte loops explicit. Objections?

Reply:
Yes, let's remove the check. But you may want to check for the other PR to be merged and you can rebase it to it.

Reply:
Done.